app/controllers/suggestions.rb

```diff
@@ -0,0 +1,142 @@
+get '/suggestions' do
+  @suggestions = Suggestion.all
+  @sorted_suggestions = @suggestions.sort_by{|suggestion| suggestion.votes.count}.reverse
+  @user = User.find(session[:id])
+
+  erb :'/suggestions/index'
+end
+
+get '/suggestions/new' do
+  erb :'/suggestions/new'
+end
+
+post '/suggestions' do
+  @suggestion = Suggestion.new(params[:suggestion])
+  current_user = User.find(session[:id])
+  @suggestion.user = current_user
+  if @suggestion.save
+    redirect '/suggestions'
+  else
+    erb :'/suggestions/new'
+  end
+end
+
+get '/suggestions/:id' do
+  @suggestion = Suggestion.find(params[:id])
+  erb :'/suggestions/show'
+end
+
+get '/suggestions/:id/edit' do
+  @suggestion = Suggestion.find(params[:id])
+  if @suggestion.user == current_user
+    erb :'/suggestions/edit'
+  else
+    @error = "You are not authorized to edit this."
+  end
+end
+
+put '/suggestions/:id' do
+  @suggestion = Suggestion.find(params[:id])
+  @suggestion.assign_attributes(params[:suggestion])
```

Comment:
Uh oh! The actual update is not protected. Remember: I can hit this update route from the command line, no need to get access to the form. Thus, protect this update like you protected the edit form.